packages/master/master.js

```diff
@@ -1610,7 +1625,9 @@ async function handleHandshake(message, socket, req, attachHandler) {
 		}
 
 		logger.verbose(`SOCKET | registered slave ${data.id} version ${data.version}`);
-		slaveConnections.set(data.id, new SlaveConnection(data, connector));
+		let slave = new SlaveConnection(data, connector);
+		slaveConnections.set(data.id, slave);
+		db.slaves.updateGui(data.id);
```

Comment:
I don't see the reason for changing this line. You don't use `slave` in a new place that necessitate it being store in a variable. The name is also misleading.